src/bio/bio_xstream.c

```diff
@@ -73,6 +76,85 @@ struct bio_nvme_data {
 static struct bio_nvme_data nvme_glb;
 uint64_t vmd_led_period;
 
+static int64_t
+hexstr_to_int(const char *src)
+{
+	uint64_t val;
+
+	val = strtoul(src, NULL, 16);
+	if (errno == ERANGE) {
+		D_ERROR("%s hex string out of range\n", src);
+		return -1;
+	}
+
+	D_INFO("hex2dec val: %lu\n", val);
+
+	return (int64_t)val;
+}
+
+static int64_t
+traddr_to_bus_id(char *src)
+{
+	char		*tok;
+	int		 i;
+
+	for (i = 0; i < 2; i++) {
+		tok = strtok(src, ":");
+		if (tok == NULL) {
+			D_ERROR("pci address field %d empty\n", i);
+			return -1;
+		}
+		D_INFO("pci address field %d: %s\n", i, tok);
+	}
+
+	return hexstr_to_int(tok);
+}
+
+static bool
+hotplug_filter_fn(const struct spdk_pci_addr *addr)
+{
+	char		str[128];
+	int64_t		low;
+	int64_t		high;
+	int64_t		bus_id;
+
+	if (spdk_pci_addr_fmt(str, sizeof(str), addr) != 0) {
+		D_ERROR("unable to format pci address\n");
+		return false;
+	}
+
+	if ((!g_hotplug_bus_id_low) || (!g_hotplug_bus_id_high))
+		return false;
```

Comment:
I think filter function returning 'true' means the device should be included, 'false' means the device should be excluded, right? so I think filter function should return 'true' if there is no filter configured.